### ISSUE TITLE
[CT-1322] [Feature] Enable setting execution_project in the dbt_project.yml

### ISSUE THREAD

**Issue by matt-winkler** [dbt-labs/dbt-bigquery#343]
### Is this your first time submitting a feature request?

- [X] I have read the [expectations for open source contributors](https://docs.getdbt.com/docs/contributing/oss-expectations)
- [X] I have searched the existing issues, and I could not find an existing issue for this feature
- [X] I am requesting a straightforward extension of existing dbt-bigquery functionality, rather than a Big Idea better suited to a discussion

### Describe the feature

Similar conceptually to how we enable end users to override virtual warehouses at the folder / model level for Snowflake, is it possible to enable setting the `execution_project` on bigquery in the dbt_project.yml?

Seems related to https://github.com/dbt-labs/dbt-bigquery/issues/67

### Describe alternatives you've considered

Multiple target profiles in dbt-core, multiple projects in dbt Cloud.

### Who will this benefit?

Bigquery users who want more fine-grained control over billing resources associated with runs.

### Are you interested in contributing this feature?

Yes

**Comment by dbeatty10** [comment]
@matt-winkler thanks for proposing this feature! Can definitely see the value in fine-grained control over billing resources💰 

### TL;DR
I'm not aware of a simple way we could implement the proposed feature.

### Details
From what I can tell, the current [`execution_project` configuration](https://docs.getdbt.com/reference/warehouse-setups/bigquery-setup#execution-project) within `profiles.yml` takes affect only at connection time and applies to all queries that use that connection. It ultimately includes the value as a parameter to [`google.cloud.bigquery.Client()`](https://github.com/dbt-labs/dbt-bigquery/blob/11478b8d859b228f1f530eeec71244dc7fc88226/dbt/adapters/bigquery/connections.py#L334-L339) when creating a client connection (see BiqQuery docs [here](https://github.com/googleapis/python-bigquery/blob/eb49873176dee478617eb50472d44703abca53b5/google/cloud/bigquery/client.py#L170-L177)).

For dbt-snowflake, the warehouse can also be [specified at connection time](https://github.com/dbt-labs/dbt-snowflake/blob/6a1647e9361c4f941b4324694133a6086a239a9d/dbt/adapters/snowflake/connections.py#L287-L300
) using the `warehouse` specified in `profiles.yml`.

But in contrast, it can be [customized on a per model basis](https://github.com/dbt-labs/dbt-snowflake/blob/21b52127e7d221db8b92114aae066fb8a7151bba/dbt/adapters/snowflake/impl.py#L73-L84) (using the `snowflake_warehouse` config). It will execute a query like this as a pre-hook of each model:
```sql
use warehouse my_purple_tshirt
```

To take the same approach in dbt-bigquery, we'd need an analagous query to execute. Do you know of anything like the following in BigQuery?
```
use project my_billing_project
```

If not, I don't know a direct approach to implement this.

**Comment by dbeatty10** [comment]
Maybe something like [this](https://cloud.google.com/bigquery/docs/multi-statement-queries#set_system_variable)?
```sql
SET @@dataset_project_id = 'MyProject';
```

**Comment by jtcohen6** [comment]
This is a similar ask to the one made in https://github.com/databricks/dbt-databricks/issues/59. We don't _really*_ expose node-level information when creating / reusing connections. We'd need that in order to pull an `execution_project` config off the node, and use it to instantiate a different connection.

*We _sorta_ do this, in order to support query comments/headers containing node-specific info, but [that code is pretty gnarly](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/adapters/base/query_headers.py).

@dbeatty10 As you mention, Snowflake lets us cheat around this via the `use warehouse` meta-query, without actually needing to alter the connection itself.
